Components/Sprite.js

```diff
@@ -0,0 +1,33 @@
+import React, { useState, useRef, useEffect } from "react";
+import { StyledSpriteContainer, StyledTileContainer, StyledSpriteImage } from "./styles";
+
+const Sprite = (props) => {
+  const { src, tile, scale, framesPerSprite, totalSprites } = props;
+  const [sprite, setSprite] = useState(0);
+  const requestIdRef = useRef();
+
+  useEffect(() => {
+    requestIdRef.current = requestAnimationFrame(animate);
```

Comment:
requestAnimationFrame returns a request Id which is stored as a reference via the useRef hook